### DISCUSSION TITLE
Detect expired auth token using SDK

### DISCUSSION THREAD

**Discussion by elliotsoomro** [pocketbase/pocketbase#1482] [Q&A] [answered]
How to detect expired auth token and trigger logout using the JavaScript SDK? I am not receiving unathorized errors from pocketbase. Parsing the token on the client does not account for if the token was invalidated.

**Comment by ganigeorgiev** [comment, marked as answer]
For a loosely check you can use `pb.authStore.isValid` getter. It returns `true` if the token in the local AuthStore is set and its `exp` claim is still valid.

For a server-side check, you can send a `authRefresh()` request. On valid token it will return up-to-date user data and token (it will automatically update the `pb.authStore`); on failure - it will throw an error:

```js
try {
    pb.authStore.isValid && await pb.collection('users').authRefresh();
} catch (e) {
    pb.authStore.clear(); // after this call pb.authStore.isValid will be false
}
```

**Reply by maxwell-oroark** [reply]
Is there a reason why pb API doesn't return 401s or similar for invalid/expired tokens?

**Reply by ganigeorgiev** [reply]
@maxwell-oroark I think the main reasoning behind this was to minimize the unnecessary errors in case a route is public or it is serving different content based on the auth state (_this could be done even with the API rules by something like `@request.auth.id != "" || status = "public"`_) which would require the user to manually call `pb.authStore.clear()` before making the request in case for example the token secret has changed between requests.
But nothing is set in stone yet and the behavior may change in the future (_I've added it in my todo to think a little more on it_).
